routes/users-new.js

```
@@ -0,0 +1,128 @@
+const express = require("express");
+const router = express.Router();
+
+const bcrypt = require("bcrypt");
+
+const methodNotAllowed = require("../errors/methodNotAllowed");
+const serverError = require("../errors/serverError");
+const sendEmail = require("../services/sendEmail");
+const { auth, authErrorHandler } = require("../middleware/auth");
+
+const TempUser = require("../models/tempUser");
+const User = require("../models/user");
+
+
+router.use(auth);
+router.use(authErrorHandler);
+
+router.post("/", async (req, res, next) => {
+
+  const emails = req.body.emails;
+  const isGrad = req.body.isGrad;
+
```

This should throw a 400 error if "emails" or "isGrad" is missing from the request's JSON object. Or if emails is not an array.

---

Handled for:
- No "emails" property attached to request
- "emails" array is empty
- "emails" object is not an array
- No "isGrad" property attached to request
- "isGrad" value is not a boolean